bin/reth/src/config.rs

```diff
@@ -29,12 +29,13 @@ impl Config {
         chain_id: u64,
         genesis_hash: H256,
         disable_discovery: bool,
+        envs: &Option<Vec<NodeRecord>>,
     ) -> NetworkConfig<ProviderImpl<DB>> {
         let peer_config = reth_network::PeersConfig::default()
             .with_trusted_nodes(self.peers.trusted_nodes.clone())
             .with_connect_trusted_nodes_only(self.peers.connect_trusted_nodes_only);
         NetworkConfig::builder(Arc::new(ProviderImpl::new(db)), rng_secret_key())
-            .boot_nodes(mainnet_nodes())
+            .set_bootnodes(envs)
```

Comment:
Sorry, maybe I was not clear.

```suggestion
            .boot_nodes(bootnodes.unwrap_or_else(|| mainnet_nodes()))
```